src/IECoreScene/PrimitiveInterpolator.cpp

```diff
@@ -71,7 +73,49 @@ PrimitivePtr interpolatePrimitive( const Primitive *y0, const Primitive *y1, dou
 			it->second.interpolation == namedPrimitiveVariable.second.interpolation
 		)
 		{
-			ObjectPtr interpolatedData = linearObjectInterpolation( namedPrimitiveVariable.second.data.get(), it->second.data.get(), x );
+			const PrimitiveVariable &y0PrimVar = namedPrimitiveVariable.second;
+			const PrimitiveVariable &y1PrimVar = it->second;
+
+			const Data *y0Sample = y0PrimVar.data.get();
+			const Data *y1Sample = y1PrimVar.data.get();
+
+			size_t y0Size = IECore::size( y0Sample );
+			size_t y1Size = IECore::size( y1Sample );
+
+			if( y0Size != y1Size )
+			{
+				msg(
+					MessageHandler::Level::Error, "interpolatePrimitive", boost::str(
+						boost::format( "primitive variable '%s' data array size changes between samples. ( sample 0 size: %i sample 1 size: %i, alpha: %f )" ) %
+							it->first %
+							y0Size %
+							y1Size %
+							x
+					)
+				);
+				return nullptr;
+			}
+
+			if( y0PrimVar.indices && y1PrimVar.indices )
```

Comment:
Should we also cover the case where one primvar has indices and the other doesn't?